### PR TITLE
Text Analytics - Add Opinion mining

### PR DESCRIPTION
**Pull request by suhas92**
This  PR adds opinion mining option to text analytics class.

### CODE REVIEW

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala [hunk 1]
```diff
@@ -194,19 +207,25 @@ class TextSentimentV4(override val uid: String)
...
-                                   options: Option[TextAnalyticsRequestOptions],
                                    input: Seq[String],
                                    lang: Seq[String]): TAResponseV4[SentimentScoredDocumentV4] = {
+
     val documents = (input, lang, lang.indices).zipped.map { (doc, lang, i) =>
       new TextDocumentInput(i.toString, doc).setLanguage(lang)
     }.asJava
 
-    val response = client.analyzeSentimentBatchWithResponse(documents, options.orNull, Context.NONE).getValue
+    val options = new AnalyzeSentimentOptions()
```
Comment (suhas92) [inline cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala:222]: if this pattern is good, i ll update other functions as well

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDKSchemasV4.scala [hunk 2]
```diff
@@ -336,12 +332,12 @@ object SDKConverters {
       entity.getEntities.getWarnings.asScala.toSeq.map(fromSDK))
   }
 
-  def toSDK(textAnalyticsRequestOptionsV4: TextAnalyticsRequestOptionsV4): TextAnalyticsRequestOptions = {
-    new TextAnalyticsRequestOptions()
-      .setModelVersion(textAnalyticsRequestOptionsV4.modelVersion)
-      .setIncludeStatistics(textAnalyticsRequestOptionsV4.includeStatistics)
-      .setServiceLogsDisabled(textAnalyticsRequestOptionsV4.disableServiceLogs)
-  }
+//  def toSDK(textAnalyticsRequestOptionsV4: TextAnalyticsRequestOptionsV4): TextAnalyticsRequestOptions = {
```
Comment (suhas92) [inline cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDKSchemasV4.scala:335]: will clean up

File: cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala [hunk 3]
```diff
@@ -244,6 +241,25 @@ class SentimentAnalysisSuiteV4 extends TestBase with DataFrameEquality with Text
...
+      .collect()
+    assert(replies(0).schema(0).name == "output")
+    df.printSchema()
+    df.show()
+    val fromRow = SentimentResponseV4.makeFromRowConverter
+
+    replies.foreach { row =>
+
+      val outResponse = fromRow(row.getAs[GenericRowWithSchema]("output"))
+      println("Target text - " + outResponse.result.head.get.sentences.head.opinions.get.head.target.text)
```
Comment (suhas92) [inline cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:258]: I want to know if there is a good pattern of asserting nested values

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala [hunk 4]
```diff
@@ -153,14 +168,13 @@ class LanguageDetectionV4(override val uid: String)
...
   override def invokeTextAnalytics(client: TextAnalyticsClient,
-                                   options: Option[TextAnalyticsRequestOptions],
                                    input: Seq[String],
                                    hints: Seq[String]): TAResponseV4[DetectedLanguageV4] = {
     val documents = (input, hints, input.indices).zipped.map { (doc, hint, i) =>
       new DetectLanguageInput(i.toString, doc, hint)
     }.asJava
 
-    val response = client.detectLanguageBatchWithResponse(documents, options.orNull, Context.NONE).getValue
+    val response = client.detectLanguageBatchWithResponse(documents, null, Context.NONE).getValue
```
Comment (mhamilton723) [inline cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala:177]: This is where you need to plumb in the options

File: cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala (hunk 1 above)
Comment (mhamilton723) [reply cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala:222]: LGTM for first pass

File: cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala (hunk 3 above)
Comment (mhamilton723) [reply cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:258]: Given that the services responses will change, you can just write some simple checks that its within a range of appropriate numbers or something like that. Equality testing is often too brittle for the services

**Review by mhamilton723 — CHANGES_REQUESTED** [review]
Looks great! Left a few comments! Thanks for your hard work :)

File: cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala [hunk 5]
```diff
@@ -244,6 +241,25 @@ class SentimentAnalysisSuiteV4 extends TestBase with DataFrameEquality with Text
...
+      println("Target sentiment - " + outResponse.result.head.get.sentences.head.opinions.get.head.target.sentiment)
```
Comment (mhamilton723) [inline cognitive/src/test/scala/com/microsoft/ml/spark/cognitive/split1/TextAnalyticsSDKSuite.scala:259]: this isnt actually checking anything

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**